tools/toolchains/gcc.py

```diff
@@ -166,7 +166,11 @@ def get_dep_option(self, object):
         return ["-MD", "-MF", dep_path]
 
     def get_compile_options(self, defines, includes):
-        return ['-D%s' % d for d in defines] + ['@%s' % self.get_inc_file(includes)]
+        opts = ['-D%s' % d for d in defines] + ['@%s' % self.get_inc_file(includes)]
+        config_header = self.get_config_header()
+        if config_header is not None:
+            opts = opts + ['-include', config_header]
```

Comment:
The differerce is that declarations in a `-include`d file are heeded where as in a `-imacros`ed file they are not.